generators/php/base/src/context/PhpAttributeMapper.ts

```diff
@@ -148,8 +148,25 @@ export class PhpAttributeMapper {
             }
             case "enumString":
                 return php.codeblock("'string'");
+            case "literal":
+                return this.getLiteralArgument(type);
             default:
                 assertNever(type.internalType);
         }
     }
+
+    private getLiteralArgument(type: php.Type) {
+        if (type.internalType.type !== "literal") {
+            throw new Error("Expected provided type to be a literal.");
+        }
```

Comment:
I don't think we need to throw here - we've already narrowed the type in the `switch` above, so we should be able to narrow the type (rather than use `php.Type`). That will prevent us from needing to `throw` here.